### PR TITLE
Update core dependencies

### LINKED ISSUE #1292: Upgrade dependencies (risc0, reth, alloy etc)
We can only upgrade some of the traits through forks. The first one's that I can think of:
- `risc0-*`
- `alloy-primitives`, `reth-primitives` --> basically anything that goes to non native EVM and other crates used inside guest codes.

### LINKED ISSUE #1523: Remove fn convert_u64_to_u256
Because modern version of alloy U256 implements U256::from for u64.

### LINKED ISSUE #1529: Use ethereum encoding where applicable for state
In #1518 we saw that with updates from external dependencies, bincode serialization on things we commit to the state sometimes change, enforcing us to implement weird workarounds.

I am going to admit that we were in the wrong by not implementing such a system for everything we write to state / accessory state:

- No encoding raw bytes StateValue / StateVec / StateMap
- Using existing ethereum encoding (e.g. RLP) after reading from such State*

For example, the serialization of header structure in alloy::consensus changes, and it will change, because it's not something you should commit to. It's for RPC implementations...

However, the encode/decode function of the header on the other hand, won't ever change, and always has to be backwards compatible. Because no one will go back and change what's committed in Ethereum. Those encode / decode functions abide by Ethereum spec.

If we were to implement such a thing, we wouldn't worry about updating reth or alloy, because we would always pull the encoded bytes from state and decode in to the new versions struct. (BTW we can have a "CustomCodec" which calls a trait function of a type, which we hook to these encode/decode functions. BUT WE CAN'T IMPLEMENT TRAITS FOR EXTERNAL DEPENDENCIES SO A LITTLE BIT OF UGLIER CODE AND HANDLING ENCODING FOR STATE* MANUALLY EVERYWHERE IS FINE)

This will be very very very hard to implement on testnet probably, however, we can do it right before mainnet, and worry about updating the testnet later.

### PR DESCRIPTION
**Pull request by ercecan**
Updates
Reth: 1.0.4 -> 1.1.0
Revm: 12.1 -> 14.0.3
revm-primitives: 8 -> 10.0.0
alloy: 0.2.1 -> 0.4.2
alloy-core deps 0.7.x -> 0.8.x

- Uses rlp encoding for storage, because with rlp encoding we will have future proof serializations and deserializations

Closes #1523 
Closes #1292 
Closes #1529

### CODE REVIEW

File: crates/evm/src/hooks.rs
Comment (ercecan) [inline crates/evm/src/hooks.rs:1]: combine

File: crates/evm/src/hooks.rs
Comment (ercecan) [inline crates/evm/src/hooks.rs]: combine

File: crates/evm/src/tests/utils.rs
Comment (kpp) [inline crates/evm/src/tests/utils.rs]: Why not also import alloy_primitives::U256?

File: crates/evm/src/tests/tx_tests.rs
Comment (kpp) [inline crates/evm/src/tests/tx_tests.rs:75]: It is `SYSTEM_SIGNATURE`.

File: crates/evm/src/tests/sys_tx_tests.rs
Comment (kpp) [inline crates/evm/src/tests/sys_tx_tests.rs:868]: Well, you can replace it with `b256!("deaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddead");`

File: crates/evm/src/tests/queries/evm_call_tests.rs
Comment (kpp) [inline crates/evm/src/tests/queries/evm_call_tests.rs]: TODO

File: crates/evm/src/tests/queries/evm_call_tests.rs
Comment (kpp) [inline crates/evm/src/tests/queries/evm_call_tests.rs]: TODO

File: crates/evm/src/tests/hooks_tests.rs
Comment (kpp) [inline crates/evm/src/tests/hooks_tests.rs]: SYSTEM_SIGNATURE

File: crates/evm/src/rpc_helpers/responses.rs
Comment (kpp) [inline crates/evm/src/rpc_helpers/responses.rs]: + B256

File: crates/evm/src/query.rs
Comment (kpp) [inline crates/evm/src/query.rs:183]: Do we need to clone here?

File: crates/evm/src/evm/system_events.rs
Comment (kpp) [inline crates/evm/src/evm/system_events.rs:9]: https://github.com/alloy-rs/core/pull/810

File: crates/evm/src/evm/db.rs
Comment (kpp) [inline crates/evm/src/evm/db.rs:78]: Is it really required?

File: crates/evm/Cargo.toml
Comment (kpp) [inline crates/evm/Cargo.toml]: TODO?

File: crates/ethereum-rpc/src/gas_price/gas_oracle.rs
Comment (kpp) [inline crates/ethereum-rpc/src/gas_price/gas_oracle.rs]: https://github.com/chainwayxyz/citrea/issues/1523

File: Cargo.lock
Comment (kpp) [inline Cargo.lock:134]: Please update deps to try to get rid of this duplicate.

File: Cargo.lock
Comment (kpp) [inline Cargo.lock:8909]: https://docs.rs/crate/serde_with/latest/features

File: Cargo.lock
Comment (kpp) [inline Cargo.lock:8936]: Please update deps to try to get rid of tower 0.4 duplicate.

File: crates/evm/src/tests/tx_tests.rs
Comment (ercecan) [reply crates/evm/src/tests/tx_tests.rs:75]: ~~cannot do a const system signature anymore because the Signature fields are now private and there is no const fn to create signature~~
Just saw your pr, gotta wait for a release tho

Comment (ercecan) [reply crates/evm/src/tests/tx_tests.rs:75]: But I can do lazy_static

File: crates/evm/src/tests/sys_tx_tests.rs
Comment (ercecan) [reply crates/evm/src/tests/sys_tx_tests.rs:868]: I tried (with bytes macro because hash is bytes) however I kept getting this error because of the macro so I let it go:
```
error[E0080]: evaluation of constant value failed
   --> /Users/erce/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hex-literal-0.4.1/src/lib.rs:17:24
    |
17  |             0..=127 => panic!("Encountered invalid ASCII character"),
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Encountered invalid ASCII character', /Users/erce/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hex-literal-0.4.1/src/lib.rs:17:24
    |
note: inside `hex_literal::next_hex_char`
   --> /Users/erce/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hex-literal-0.4.1/src/lib.rs:17:24
    |
17  |             0..=127 => panic!("Encountered invalid ASCII character"),
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `hex_literal::next_byte`
   --> /Users/erce/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hex-literal-0.4.1/src/lib.rs:30:30
    |
30  |     let (half2, pos) = match next_hex_char(string, pos) {
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `alloy_primitives::hex_literal::len`
   --> /Users/erce/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hex-literal-0.4.1/src/lib.rs:46:40
    |
46  |         while let Some((_, new_pos)) = next_byte(strings[i], pos) {
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `sys_tx_tests::test_upgrade_light_client::STATIC_BYTES::LEN`
   --> crates/evm/src/tests/sys_tx_tests.rs:608:9
    |
608 |         bytes!("0xdeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddead").unwrap()
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `bytes` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
   --> crates/evm/src/tests/sys_tx_tests.rs:608:9
    |
608 |         bytes!("0xdeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddeaddead").unwrap()
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this note originates in the macro `$crate::hex` which comes from the expansion of the macro `bytes` (in Nightly builds, run with -Z macro-backtrace for more info)

```

File: crates/evm/src/tests/hooks_tests.rs
Comment (ercecan) [reply crates/evm/src/tests/hooks_tests.rs]: I wrote above

File: crates/evm/src/rpc_helpers/responses.rs
Comment (ercecan) [reply crates/evm/src/rpc_helpers/responses.rs:3]: no

File: crates/evm/src/query.rs
Comment (ercecan) [reply crates/evm/src/query.rs:183]: yes as it is also used below

File: crates/evm/src/evm/system_events.rs
Comment (ercecan) [reply crates/evm/src/evm/system_events.rs:9]: ohh nice one ignore my previous comments about fn not being const :D

Comment (ercecan) [reply crates/evm/src/evm/system_events.rs:9]: I guess we still gotta wait for a release

File: crates/evm/src/tests/tx_tests.rs
Comment (ercecan) [reply crates/evm/src/tests/tx_tests.rs:75]: > Just saw your pr, gotta wait for a release tho

I tried with the rev after your pr merged and it broke a lot of things, we really gotta wait for new reth release for that unfortunately, or a port pr

File: crates/evm/src/evm/db.rs
Comment (ercecan) [reply crates/evm/src/evm/db.rs:78]: Unfortunately, because of this field in AccountOverride of Alloy:
```rust
pub state_diff: Option<B256HashMap<B256>>,
```

Comment (kpp) [reply crates/evm/src/evm/db.rs:78]: How about we use type B256HashMap<B256>?

File: crates/primitives/Cargo.toml
Comment (eyusufatik) [inline crates/primitives/Cargo.toml]: remove native feature.

File: crates/evm/Cargo.toml
Comment (eyusufatik) [inline crates/evm/Cargo.toml]: do we still need serde_with even after implementing the rlp stuff?

File: crates/evm/src/evm/system_events.rs
Comment (eyusufatik) [inline crates/evm/src/evm/system_events.rs:81]: craete issue about system signature const fn problem

File: crates/evm/src/rpc_helpers/tracing_utils.rs
Comment (eyusufatik) [inline crates/evm/src/rpc_helpers/tracing_utils.rs]: return Unsupported for these

**Review by eyusufatik — CHANGES_REQUESTED** [review]
open issue for unnecessary double hashing for rlp decode header

File: crates/evm/src/evm/system_events.rs
Comment (eyusufatik) [reply crates/evm/src/evm/system_events.rs:81]: https://github.com/chainwayxyz/citrea/issues/1622

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**
